mage_ai/data_preparation/repo_manager.py

```diff
@@ -293,23 +293,28 @@ def get_variables_dir(
     return variables_dir
 
 
-project_uuid = None
-try:
-    with get_metadata_path() as f:
-        config = yml.load(f) or {}
-        project_uuid = config.get('project_uuid')
-except Exception:
-    pass
+def set_project_uuid_from_metadata():
+    global project_uuid
+    try:
+        with open(get_metadata_path(), 'r', encoding='utf-8') as f:
+            config = yml.load(f) or {}
+            project_uuid = config.get('project_uuid')
+    except Exception:
+        pass
 
 
-def update_project_uuid():
+def init_project_uuid():
```

Comment:
Can we add another check here? If `get_repo_config().project_uuid` exists, we don't need to set the new uuid, right?

Reply:
yeah the idea is that the `project_uuid` global variable will contain the value from `get_repo_config().project_uuid`

Reply:
I don't know whether there could be race conditions if multiple servers are started since we allow user to start multiple servers.

Reply:
oh, just saw you added the check here. so it should be good.

Reply:
I added another check

Reply:
sorry, i was looking at it again, and it made sense to add the 2nd check